src/main/groovy/org/burnmydread/SimpleRandom.groovy

```diff
@@ -12,12 +12,36 @@ import java.nio.ByteBuffer
  * The intent is to have a easy to use Secure RNG.
  */
 class SimpleRandom {
+    /* MP COMMENT
+     * Couple of points here:
+     * 1. This throws an IOException that needs to be handled, so at a minimum you'll need
+     *    a constructor for this class, although IO should generally not be done in a constructor
+     *    either.  Even better would be have an init method that initializes its state, but
+     *    that makes the API a little harder to use.
+     * 2. Check for the existence of /dev/urandom first - preferable to use this on Linux. 
+     *    FreeBSD only has /dev/random, so fallback to using that.  
+     * 3. Readers are for text-based input, whereas I think of /dev/urandom as a byte stream,
+     *    so you should use a Java InputStream. Not sure about using a BufferedInputStream either -
+     *    I don't think you want that here.  I think you just want to read unbuffered, but
+     *    I could be thinking incorrectly about that.
+     */
     private static final BufferedReader devRandom = new BufferedReader(new FileReader('/dev/random'))
-
+    
 
     void nextBytes(byte[] bytes) {
+        /* MP COMMENT
+         * Using a char array is not useful here and technically this line is wrong - byte is a single
+         * byte whereas char is 2 bytes in Java. If you switch to InputStream rather than reader you'll
+         * just deal with byte[], as is appropriate here.
+         */
         char[] out = new char[bytes.length]
         int index = 0
+        /* MP COMMENT
+         * How were you measuring that you were getting poor entry after the first 32 bytes?
+         * I'm not clear on how I would know that unless I was just getting repeated data.
+         * In which case, using /dev/urandom on Linux should fix that problem and you can
+         * then just do a single read of the desired size.
+         */
```

Comment:
The tests display the raw results from the two list I used to do an initial check that we are getting somewhat ok values back. What I was seeing was the first 32 bytes where varied in their values. But the values after than were all either 0 or 8. This may be related to using /dev/random rather than /dev/urandom.